### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by SmirnovEgorRu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+```
+
+# Scikit-learn patching
+
+| *Speedups of daal4py powered Scikit-learn over the original Scikit-learn, 28 cores, 1 thread/core* |
+|:--:|
+| ![](doc/IDP%20scikit-learn%20accelearation%20compared%20with%20stock%20scikit-learn.png) |
+| *technical details: FPType: float32; HW: Intel(R) Xeon(R) Platinum 8276L CPU @ 2.20GHz, 2 sockets, 28 cores per socket; SW: scikit-learn 0.22.2, Intel® DAAL (2019.5), Intel® Distribution Of Python (IDP) 3.7.4; Details available in the article https://medium.com/intel-analytics-software/accelerate-your-scikit-learn-applications-a06cacf44912* |
+
+daal4py patching will affect performance for specific Scikit-learn functionality listed below. In some cases daal4py can fallback into stock Scikit-learn, if there are some unsupported parameters (these limitations described below). If the patching doesn't cover your scenarios - please, submit an issue on GitHub.
```
Comment (outoftardis) [inline README.md:81]:
```suggestion
daal4py patching affects performance of specific Scikit-learn functionality listed below. In cases when unsupported parameters are used, daal4py fallbacks into stock Scikit-learn. These limitations described below. If the patching does not cover your scenarios, [submit an issue on GitHub](https://github.com/IntelPython/daal4py/issues).
```

File: README.md [hunk 2]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+
+Already available in 2020.3 release:
```
Comment (outoftardis) [inline README.md:83]:
```suggestion
Scenarios that are already available in 2020.3 release:
```

File: README.md [hunk 3]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+|Task|Functionality|Parameters support|Data support|
+|:---|:------------|:-----------------|:-----------|
+|Classification|**SVC**|All parameters except `kernel` = 'poly' and 'sigmoid'|No limitations|
+||**RandomForestClassifier**|All parameters except `warmstart` = True and `cpp_alpha` != 0, `criterion` != 'gini' | Multi-output and sparse data aren't supported |
+||**KNeighborsClassifier**|Supported `metric` = 'euclidean' and `minkowski` with `p` = 2. Other parameters are fully supported | Multi-output and sparse data aren't supported |
+||**LogisticRegression / LogisticRegressionCV**|Supported `solver` = 'lbfgs' and 'newton-cg', `penalty` = 'l2' and 'none', `class_weight` = None. Other parameters are fully supported | Dense data |
+|Regression|**RandomForestRegressor**|All parameters except `warmstart` = True and `cpp_alpha` != 0, `criterion` != 'mse' | Sparse data is't supported |
```
Comment (PetrovKP) [inline README.md:90]:
```suggestion
|Regression|**RandomForestRegressor**|All parameters except `warmstart` = True and `cpp_alpha` != 0, `criterion` != 'mse' | Multi-output and sparse data aren't supported |
```

File: README.md [hunk 4]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+||**LinearRegression**|All parameters except `normalize` != False. | Dense data, # observations should be >= # features |
+||**Ridge**|All parameters except `normalize` != False and `solver` != 'auto' | Dense data, # observations should be >= # features |
```
Comment (outoftardis) [inline README.md:92]:
```suggestion
||**Ridge**|All parameters except `normalize` != False and `solver` != 'auto'. | Only dense data is supported, `#observations` should be >= `#features`. |
```

File: README.md [hunk 5]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+
+Already available in 2020.3 release:
+|Task|Functionality|Parameters support|Data support|
+|:---|:------------|:-----------------|:-----------|
+|Classification|**SVC**|All parameters except `kernel` = 'poly' and 'sigmoid'|No limitations|
+||**RandomForestClassifier**|All parameters except `warmstart` = True and `cpp_alpha` != 0, `criterion` != 'gini' | Multi-output and sparse data aren't supported |
+||**KNeighborsClassifier**|Supported `metric` = 'euclidean' and `minkowski` with `p` = 2. Other parameters are fully supported | Multi-output and sparse data aren't supported |
+||**LogisticRegression / LogisticRegressionCV**|Supported `solver` = 'lbfgs' and 'newton-cg', `penalty` = 'l2' and 'none', `class_weight` = None. Other parameters are fully supported | Dense data |
+|Regression|**RandomForestRegressor**|All parameters except `warmstart` = True and `cpp_alpha` != 0, `criterion` != 'mse' | Sparse data is't supported |
+||**LinearRegression**|All parameters except `normalize` != False. | Dense data, # observations should be >= # features |
```
Comment (outoftardis) [inline README.md:91]:
```suggestion
||**LinearRegression**|All parameters except `normalize` != False. | Only dense data is supported, `#observations` should be >= `#features`. |
```

File: README.md [hunk 6]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+||**Ridge**|All parameters except `normalize` != False and `solver` != 'auto' | Dense data, # observations should be >= # features |
+||**ElasticNet**|All parameters are supported| Multi-output and sparse data aren't suppported. # observations should be >= # features |
+||**Lasso**|All parameters are supported| Multi-output and sparse data aren't suppported. # observations should be >= # features |
+|Clustering|**KMeans**|All parameters except `precompute_distances` | No limitations |
```
Comment (outoftardis) [inline README.md:95]:
```suggestion
|Clustering|**KMeans**|All parameters except `precompute_distances`. | No limitations. |
```

File: README.md (hunk 1 above)
Comment (PetrovKP) [inline README.md:81]: Need to add information about the verbose mode before description about Sklearn parameters.

File: README.md [hunk 7]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+||**DBSCAN**|Supported `metric` = 'euclidean' and 'minkowski' with p='2', 'algorithm'='brute. Other parameters are fully supported | Dense data only |
```
Comment (outoftardis) [inline README.md:96]:
```suggestion
||**DBSCAN**|Supported `metric` = 'euclidean' and 'minkowski' with p='2', 'algorithm'='brute. Other parameters are fully supported. | Only dense data is supported. |
```

File: README.md [hunk 8]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+|Dimensionality reduction|**PCA**|All parameters except `svd_solver` != 'full' | No limitations |
```
Comment (outoftardis) [inline README.md:97]:
```suggestion
|Dimensionality reduction|**PCA**|All parameters except `svd_solver` != 'full'. | No limitations. |
```

File: README.md [hunk 9]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+| *Speedups of daal4py powered Scikit-learn over the original Scikit-learn, 28 cores, 1 thread/core* |
+|:--:|
+| ![](doc/IDP%20scikit-learn%20accelearation%20compared%20with%20stock%20scikit-learn.png) |
+| *technical details: FPType: float32; HW: Intel(R) Xeon(R) Platinum 8276L CPU @ 2.20GHz, 2 sockets, 28 cores per socket; SW: scikit-learn 0.22.2, Intel® DAAL (2019.5), Intel® Distribution Of Python (IDP) 3.7.4; Details available in the article https://medium.com/intel-analytics-software/accelerate-your-scikit-learn-applications-a06cacf44912* |
+
+daal4py patching will affect performance for specific Scikit-learn functionality listed below. In some cases daal4py can fallback into stock Scikit-learn, if there are some unsupported parameters (these limitations described below). If the patching doesn't cover your scenarios - please, submit an issue on GitHub.
+
+Already available in 2020.3 release:
+|Task|Functionality|Parameters support|Data support|
+|:---|:------------|:-----------------|:-----------|
```
Comment (PetrovKP) [inline README.md:85]: Don't want to add information about samples weights?

File: README.md [hunk 10]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+||**ElasticNet**|All parameters are supported| Multi-output and sparse data aren't suppported. # observations should be >= # features |
+||**Lasso**|All parameters are supported| Multi-output and sparse data aren't suppported. # observations should be >= # features |
+|Clustering|**KMeans**|All parameters except `precompute_distances` | No limitations |
+||**DBSCAN**|Supported `metric` = 'euclidean' and 'minkowski' with p='2', 'algorithm'='brute. Other parameters are fully supported | Dense data only |
+|Dimensionality reduction|**PCA**|All parameters except `svd_solver` != 'full' | No limitations |
+|Other|**train_test_split**|All parameters are supported| Dense data only |
+||**assert_all_finite**|All parameters are supported| Dense data only |
+||**pairwise_distance**|With `metric`='cosine' and 'correlation'| Dense data only |
+
+Not released yet, but available in master (will be included in future releases):
```
Comment (outoftardis) [inline README.md:102]:
```suggestion
Scenarios that are only available in the `master` branch (not released yet):
```

File: README.md [hunk 11]
```diff
@@ -59,19 +59,55 @@ with sycl_context("gpu"):
...
+
+|Task|Functionality|Parameters support|Data support|
+|:---|:------------|:-----------------|:-----------|
+|Regression|**KNeighborsRegressor**|Supported `metric` = 'euclidean' and 'minkowski' with `p` = 2. Other parameters are fullysupported | Sparse data is't supported |
```
Comment (PetrovKP) [inline README.md:106]:
```suggestion
|Regression|**KNeighborsRegressor**|Supported `metric` = 'euclidean' and 'minkowski' with `p` = 2. Other parameters are fullysupported | Multi-output and sparse data aren't supported |
```

**Comment by outoftardis** [comment]
you can choose `data is` or `data are` but please use it consistently (the first one is preferable but plural also ok)

**Comment by SmirnovEgorRu** [comment]
@outoftardis, @PetrovKP, thank you, I applied the comments.

**Review by napetrov — APPROVED** [review]

File: README.md [hunk 12]
```diff
@@ -59,135 +60,66 @@ with sycl_context("gpu"):
...
+||**LogisticRegression / LogisticRegressionCV**|Supported `solver` = 'lbfgs' and 'newton-cg', `penalty` = 'l2' and 'none', `class_weight` = None, `sample_weight` = None. Other parameters are fully supported. | Only dense data is supported. |
+|Regression|**RandomForestRegressor**|All parameters except `warmstart` = True and `cpp_alpha` != 0, `criterion` != 'mse'. | Multi-output and sparse data is not supported. |
+||**LinearRegression**|All parameters except `normalize` != False and `sample_weight` != None. | Only dense data is supported, `#observations` should be >= `#features`. |
+||**Ridge**|All parameters except `normalize` != False, `solver` != 'auto' and `sample_weight` != None. | Only dense data is supported, `#observations` should be >= `#features`. |
+||**ElasticNet**|All parameters are supported except `sample_weight` | Multi-output and sparse data is not supported, `#observations` should be >= `#features`. |
+||**Lasso**|All parameters are supported except `sample_weight`| Multi-output and sparse data is not supported, `#observations` should be >= `#features`. |
+|Clustering|**KMeans**|All parameters except `precompute_distances`. | No limitations. |
```
Comment (PetrovKP) [inline README.md:96]:
```suggestion
|Clustering|**KMeans**|All parameters except `precompute_distances` and `sample_weight` != None. | No limitations. |
```

File: README.md [hunk 13]
```diff
@@ -59,135 +60,66 @@ with sycl_context("gpu"):
...
+result = init.compute(X, daal4py.my_procid())
+d4p.daalfini()
+```
+
+# Scikit-learn patching
+
+| *Speedups of daal4py-powered Scikit-learn over the original Scikit-learn* |
+|:--:|
+| ![](doc/IDP%20scikit-learn%20accelearation%20compared%20with%20stock%20scikit-learn.png) |
+| *Technical details: FPType: float32; HW: Intel(R) Xeon(R) Platinum 8280 CPU @ 2.70GHz, 2 sockets, 28 cores per socket; SW: scikit-learn 0.23.1, Intel® oneDAl (2021.1 Beta 10)* |
```
Comment (PetrovKP) [inline README.md:80]: It seems to me this is not a well-known abbreviation
```suggestion
| *Technical details: float type: float32; HW: Intel(R) Xeon(R) Platinum 8280 CPU @ 2.70GHz, 2 sockets, 28 cores per socket; SW: scikit-learn 0.23.1, Intel® oneDAl (2021.1 Beta 10)* |
```

Comment (PetrovKP) [inline README.md:80]: Are you sure that float32 and not float64?

Comment (SmirnovEgorRu) [reply README.md]: fixed

File: INSTALL.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
+# Building from sources
```
Comment (outoftardis) [inline INSTALL.md:1]: I'd add a first-level heading for the whole file because this file contains various scenarios, and then a table of contents.

File: README.md [hunk 15]
```diff
@@ -9,26 +9,27 @@ A simplified API to Intel(R) oneAPI Data Analytics Library that allows for fast
 - [Documentation](https://intelpython.github.io/daal4py/)
 - [scikit-learn API and patching](https://intelpython.github.io/daal4py/sklearn.html#sklearn)
 - [Source Code](https://github.com/IntelPython/daal4py/tree/master/src)
+- [Building from Sources](INSTALL.md)
```
Comment (outoftardis) [inline README.md:12]: That file contains more than just building from sources, I suggest you change the alt text for this link.
You can also add this link at the end of Installation section with some explanation of what INSTALL.md contains.

File: README.md [hunk 16]
```diff
@@ -59,135 +60,66 @@ with sycl_context("gpu"):
...
+- On Windows: `set IDP_SKLEARN_VERBOSE=INFO`
+
+For example, for DBSCAN you get one of these print statements depending on which implementation is used:
+- `INFO: sklearn.cluster.DBSCAN.fit: uses Intel(R) oneAPI Data Analytics Library solver`
+- `INFO: sklearn.cluster.DBSCAN.fit: uses original Scikit-learn solver`
+
+[Read more in the documentation](https://intelpython.github.io/daal4py/sklearn.html#scikit-learn-verbose).
+
+# Building from Source
+See [Building from Sources](INSTALL.md) for details.
```
Comment (outoftardis) [inline README.md:125]: I think this should be a subsection in Installation (see my other comments in a previous review).

File: README.md (hunk 15 above)
Comment (SmirnovEgorRu) [reply README.md:12]: It contains building daal4py from sources + building documentation from sources only. So, I suppose the name covers this. Do you see anything else in INSTALL.md?

Comment (outoftardis) [reply README.md:12]: You are right. I looked through the file and the hierarchy confused me, it made it seem that the file covers different things. Sorry.

**Review by outoftardis — APPROVED** [review]

File: README.md (hunk 15 above)
Comment (SmirnovEgorRu) [reply README.md:12]: Anyway, I added different heading in INSTALL.md and added a link in "Installation". Thank you.
